src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -474,7 +483,10 @@ private void insertParameter(DetailAST ast) {
     private void insertVariable(DetailAST ast) {
         final Map<String, FinalVariableCandidate> scope = scopeStack.peek().scope;
         final DetailAST astNode = ast.findFirstToken(TokenTypes.IDENT);
-        scope.put(astNode.getText(), new FinalVariableCandidate(astNode));
+        final FinalVariableCandidate candidate = new FinalVariableCandidate(astNode);
+        // for-each variables are implicitly assigned
+        candidate.assigned = ast.getParent().getType() == TokenTypes.FOR_EACH_CLAUSE;
```

Comment:
You set `assigned`, but what about the field `alreadyAssigned`? Does it matter that this field stays `false`?

Reply:
The `alreadyAssigned` field is set to `true` when a variable is assigned that already has `assigned` set to `true` (by `determineAssignmentConditions`).

Then final variable candidates with `alreadyAssigned` set to `true` are filtered out as no longer being candidates since they have been assigned 2+ times (by `shouldRemoveFinalVariableCandidate`).

So with this change, for-each variables get
```
for (Object o : collection) { // assigned = true, alreadyAssigned = false
  o = new String("other");    // assigned = true, alreadyAssigned = true
}
```

Reply:
I see, thanks for the explanation.